security/tutorials/calico-policy.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Calico policy tutorial
+description: Learn how to create more advanced Calico network policies (namespace, allow and deny all ingress and egress).
+canonical_url: "/security/tutorials/calico-policy"
+---
+
+Calico network policies **extend** the functionalities of Kubernetes network policies. To demonstrate this, this tutorial follows a similar approach to the [Kubernetes Advanced Network Policy Tutorial]({{ site.baseurl }}/security/tutorials/kubernetes-policy-advanced), and implements it using Calico network policies. It not only highlights the syntactical differences between the two policy types, but also demonstrates the flexibility of Calico network policies.
+
+### Requirements
+
+- A working Kubernetes cluster and access to it using kubectl and calicoctl
+- Your Kubernetes nodes have connectivity to the public internet
+- You are familiar with [Calico NetworkPolicy]({{ site.baseurl }}/security/calico-network-policy)
+
+### Tutorial flow
+
+1. Create the namespace and NGINX service
+2. Configure default deny
+3. Allow kube-system
+4. Allow access to NGINX
+5. Clean up
+
+### 1. Create the namespace and nginx service
+
+We'll use a new namespace for this guide. Run the following commands to create the namespace and a plain NGINX service listening on port 80.
+
+```bash
+kubectl create ns advanced-policy-demo
+kubectl create deployment --namespace=advanced-policy-demo nginx --image=nginx
+kubectl expose --namespace=advanced-policy-demo deployment nginx --port=80
+```
+
+#### Verify access - allowed all ingress and egress
+
+Open up a second shell session which has `kubectl` connectivity to the Kubernetes cluster and create a busybox pod to test policy access. This pod will be used throughout this tutorial to test policy access.
+
+```bash
+kubectl run --namespace=advanced-policy-demo access --rm -ti --image busybox /bin/sh
+```
+
+This will open up a shell session inside the `busybox` pod, as shown below.
+
+```
+Waiting for pod advanced-policy-demo/access-472357175-y0m47 to be running, status is Pending, pod ready: false
+
+If you don't see a command prompt, try pressing enter.
+/ #
+```
+
+{: .no-select-button}
+
+Now from within the busybox "access" pod execute the following command to test access to the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It returns the HTML of the nginx welcome page.
+
+Still within the busybox "access" pod, issue the following command to test access to google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It returns the HTML of the google.com home page.
+
+### 2. Lock down all traffic
```

Comment:
If you do this step first then if the failsafe ports are not correctly set up then Calico can cut itself off from its datastore, which means any subsequent policy changes will not be processed.

So either the kube-system policy should be before this step, or perhaps better, use projectcalico.org/namespace != kube-system in the selector, so it only default denies things that aren't in the kube-system namespace (and then you can omit the allow-kube-system policy step).

Reply:
Done! Thank you for that suggestion!!